exporter/collector/metricsexporter.go

```diff
@@ -350,7 +350,11 @@ func (m *metricMapper) metricToTimeSeries(
 		sum := metric.Sum()
 		points := sum.DataPoints()
 		for i := 0; i < points.Len(); i++ {
-			ts := m.sumPointToTimeSeries(resource, extraLabels, metric, sum, points.At(i))
+			point := points.At(i)
+			if !intervalIsValid(point.StartTimestamp().AsTime(), point.Timestamp().AsTime()) {
+				continue
+			}
```

Comment:
Would it make more sense to move this logic into the corresponding `xPointToTimeSeries()` methods?

Reply:
I thought about that, but then we would need to check the return value of xPointToTimeSeries() anyways.

Reply:
Ya, another option is having them return `[]*TimeSeries` so `append()` will do the right thing. Up to you, just calling it out

Reply:
done